[tbb] compile error(s) if TBB_USE_DEBUG is defined

### Description

- compile error(s) if TBB_USE_DEBUG is defined

(TBB_USE_DEBUG  is defined automatically if using cmake via TBBTargets.cmake)

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

Microsoft Windows Version = "10.0.19041.1806"


### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Expected behavior

should compile

### Actual behavior

1.)
C:/msys64/mingw64/include/oneapi/tbb/profiling.h:148:36: error: cannot convert 'const char*' to 'const tbb::detail::d0::tchar*' {aka 'const wchar_t*'}
  148 |         r1::itt_set_sync_name(obj, name);

2.) (only in Qt-Projects)
C:/msys64/mingw64/include/oneapi/tbb/profiling.h:229:15: error: expected unqualified-id before ')' token
  229 |     void emit() { }

(name clash with Qt's emit)


### Repro steps

```
echo -e "#include <execution>\nint main(){}" >main.cpp
gcc -DTBB_USE_DEBUG main.cpp -ltbb12
```

### Are you willing to submit a PR?

_No response_

--- a/mingw-w64-tbb/001-fix-using-TBB-with-Debug-build-type.patch
+++ b/mingw-w64-tbb/001-fix-using-TBB-with-Debug-build-type.patch
@@ -1,0 +1,11 @@
+--- a/include/oneapi/tbb/profiling.h
++++ b/include/oneapi/tbb/profiling.h
+@@ -132,7 +132,7 @@
+         r1::call_itt_notify((int)t, ptr);
+     }
+ 
+-#if (_WIN32||_WIN64) && !__MINGW32__
++#if (_WIN32||_WIN64)
+     inline void itt_set_sync_name(void* obj, const wchar_t* name) {
+         r1::itt_set_sync_name(obj, name);
+     }

--- a/mingw-w64-tbb/PKGBUILD
+++ b/mingw-w64-tbb/PKGBUILD
@@ -4,7 +4,7 @@ _realname=oneTBB
 pkgbase=mingw-w64-tbb
 pkgname=("${MINGW_PACKAGE_PREFIX}-tbb")
 pkgver=2021.7.0
-pkgrel=1
+pkgrel=2
 arch=(any)
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
 url='https://software.intel.com/en-us/oneapi/onetbb'
@@ -19,12 +19,15 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
              "${MINGW_PACKAGE_PREFIX}-ninja"
              "${MINGW_PACKAGE_PREFIX}-cc")
 source=("https://github.com/oneapi-src/${_realname}/archive/v${pkgver}/${_realname}-${pkgver}.tar.gz"
+        001-fix-using-TBB-with-Debug-build-type.patch
         004-support-mingw-aarch64.patch)
 sha256sums=('2cae2a80cda7d45dc7c072e4295c675fff5ad8316691f26f40539f7e7e54c0cc'
+            '094c556a1087563a3a5810db0fa81aed6432b3a8e91aab9a747e9a55ea262f2a'
             '0f28cdc8a90e8dc3870cd584cf9f5f683e0a35696fe4dd816d1f6b8b5ea39de5')
 
 prepare() {
   cd ${srcdir}/${_realname}-${pkgver}
+  patch -p1 -i ${srcdir}/001-fix-using-TBB-with-Debug-build-type.patch
   patch -p1 -i ${srcdir}/004-support-mingw-aarch64.patch
 }
 

